plugins/huaweicloud/src/main/java/org/eclipse/xpanse/orchestrator/plugin/huaweicloud/monitor/utils/HuaweiMonitorCache.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.orchestrator.plugin.huaweicloud.monitor.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.modules.monitor.Metric;
+import org.eclipse.xpanse.orchestrator.plugin.huaweicloud.monitor.models.HuaweiMetric;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Huawei Monitor Cache.
+ */
+@Component
+public class HuaweiMonitorCache {
+
+    private static final int DEFAULT_TIME_DIFFERENCE = 30 * 60 * 1000;
+
+    private static final Map<String, HuaweiMetric> METRIC_MAP = new ConcurrentHashMap<>();
+
+    /**
+     * Set Metric Cache.
+     */
+    public void set(String resourceId, List<Metric> metrics) {
+        if (StringUtils.isBlank(resourceId) || CollectionUtils.isEmpty(metrics)) {
+            return;
+        }
+        HuaweiMetric huaweiMetric = new HuaweiMetric(metrics);
+        METRIC_MAP.put(resourceId, huaweiMetric);
+    }
+
+    /**
+     * Determine whether the cache is empty.
+     */
+    public boolean isEmpty() {
+        return METRIC_MAP.isEmpty();
+    }
+
+    /**
+     * Get Metric Cache.
+     */
+    public List<Metric> get(String resourceId) {
+        if (StringUtils.isBlank(resourceId) || isEmpty()) {
+            return new ArrayList<>();
+        }
+        HuaweiMetric huaweiMetric = METRIC_MAP.get(resourceId);
+        if (System.currentTimeMillis() - huaweiMetric.getTime() > DEFAULT_TIME_DIFFERENCE) {
+            return new ArrayList<>();
+        }
+        return huaweiMetric.getMetrics();
+    }
+
+    /**
+     * Cache Expires (Default 30 minutes).
+     */
+    @Async("taskExecutor")
+    public void expire() {
```

Comment:
we must add MDC keys to this method. And add a log line that the old metrices are being cleared.